packages/fiori/src/SideNavigation.js

```diff
@@ -182,6 +182,12 @@ class SideNavigation extends UI5Element {
 		const treeItem = event.detail.item;
 		const item = treeItem.associatedItem;
 
+		if (!(item.getAttribute("whole-item-toggleable") === "true")) {
+			item.fireEvent("click");
```

Comment:
I think we should always fire click event when item is clicked and fire in addition "toggle" event if the item has children that are expanded collapsed without the whole-item-toggleable thing. Or, if the events are on the SideNavigation - item-click and item-toggle.